### PR TITLE
HPCC-12764 Init System - check_status rework, stop_thor fix, uslaves.sta...

### PR DESCRIPTION
**Pull request by Michael-Gardner**
The logic behind check_status was too ambiguous so I rewrote it, and subsequently made the appropriate changes anywhere it was called. Also added more debugging information into check_status so that it will be much more clear what is going wrong when we are in fact having an issue.

An error was found in stop_thor which basically made it so the branch that took care of cleaning up thorslaves would never be entered. This has been fixed and through discussion with Jake we have cleaned up how we use the uslaves.start file.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 Should be cleaner now for review.
@jakesmith Please check and ensure that the usage of uslaves.start is how you intended.

File: initfiles/componentfiles/thor/stop_thor [hunk 1]
```diff
@@ -73,24 +73,16 @@ if [ "$localthor" = "true" ]; then
...
-  # timeout of 60 seconds (in case slave busy)
-  # hard kill (called after master down anyway so going to stall)
-  if [ ! -r $instancedir/slaves ]; then
+  # we want to kill only slaves that have already been started in run_thor
+  if [ -r $instancedir/uslaves.start ]; then
     if [ ! -z ${THORPRIMARY} ]; then
         groupName=${THORPRIMARY}
     else
         groupName=${THORNAME}
     fi
```
Comment (jakesmith) [inline initfiles/componentfiles/thor/stop_thor]: the above 5 lines are no longer needed.

**Comment by jakesmith** [comment]
Other than that minor point, looks fine.

**Comment by jakesmith** [comment]
@Michael-Gardner - back to you.

**Comment by Michael-Gardner** [comment]
@richardkchapman ready for final review.

**Pull request merged**